utils/dataproc.py

```diff
@@ -391,9 +389,10 @@ def moz_dataproc_pyspark_runner(
         worker_num_local_ssds=worker_num_local_ssds,
     )
 
-    _dag_name = f"{parent_dag_name}.{dag_name}"
-
-    with models.DAG(_dag_name, default_args=default_args) as dag:
+    with (
+        dag or DagContext.get_current_dag(),
+        TaskGroup(task_group_name, default_args=default_args) as task_group,
+    ):
```

Comment:
Also does having the context managers side by side like this set the `TaskGroup`'s `dag`? If it does I'm confused as to why it wouldn't set the operator `dag`... Seems like the same problem

Reply:
> Think this is worth an Airflow issue? The `TaskGroupContext` could/should pass around the dag maybe, not sure whether that's desirable behaviour.

Yeah, it's not a huge deal, but I initially expected it to work that way.  I've submitted apache/airflow#35968 and we'll see what the Airflow maintainers think.

Reply:
> Also does having the context managers side by side like this set the `TaskGroup`'s `dag`? If it does I'm confused as to why it wouldn't set the operator `dag`... Seems like the same problem

Yes, if there's an active DAG context it will be used for the `dag` properties of both task groups and operators defined within that context unless a different DAG is manually passed in.

Currently we have three options:
1. Make sure all the DAGs where the modified utils code is being used are defined using a DAG context (`with DAG(...) as dag:`).
2. Trigger a new DAG context at the point where we need it (`with dag or DagContext.get_current_dag():`).
3. Manually pass the DAG into all the operators.

Personally I'd prefer to use DAG contexts consistently everywhere and go with option 1, but to keep this PR minimally focused on just replacing SubDAGs I choose to go with option 2 or 3 depending on whether the previous code was defining the SubDAG using a DAG context (option 2) or not (option 3).